internal/internal_task_handlers.go

```diff
@@ -852,6 +855,11 @@ ProcessEvents:
 
 		for i, event := range reorderedEvents {
 			isInReplay := reorderedHistory.IsReplayEvent(event)
+			if !isInReplay && !replayStopWatchStopped {
+				replayStopWatch.Stop()
```

Comment:
Are you 100% sure that this line is always reached while processing events?

Reply:
Added sanity check.